src/pykeen/datasets/ea/combination.py

```diff
@@ -1,26 +1,164 @@
+    if extra:
+        result.update(extra)
+    return result
+
+
+def _merge_both_mappings(
+    left: TriplesFactory,
+    right: TriplesFactory,
+    relation_offsets: torch.LongTensor,
+    # optional
+    entity_offsets: Optional[torch.LongTensor] = None,
+    extra_relations: Optional[Mapping[str, int]] = None,
+    entity_mappings: Sequence[Mapping[int, int]] = None,
+) -> Tuple[Mapping[str, int], Mapping[str, int]]:
+    # merge entity mapping
+    entity_to_id = _merge_mappings(
+        (EA_SIDE_LEFT, left.entity_to_id),
+        (EA_SIDE_RIGHT, right.entity_to_id),
+        offsets=entity_offsets,
+        mappings=entity_mappings,
+    )
+    # merge relation mapping
+    relation_to_id = _merge_mappings(
+        (EA_SIDE_LEFT, left.relation_to_id),
+        (EA_SIDE_RIGHT, right.relation_to_id),
+        offsets=relation_offsets,
+        extra=extra_relations,
+    )
+    return entity_to_id, relation_to_id
+
+
+def filter_map_alignment(
+    alignment: pandas.DataFrame,
+    left: CoreTriplesFactory,
+    right: CoreTriplesFactory,
+    entity_offsets: torch.LongTensor,
+) -> torch.LongTensor:
+    """
+    Convert dataframe with label or ID-based alignment.
+
+    :param alignment: columns: EA_SIDES
+        the dataframe with the alignment
+    :param left:
+        the triples factory of the left graph
+    :param right:
+        the triples factory of the right graph
+    :param entity_offsets: shape: (2,)
+        the entity offsets from old to new IDs
+
+    :return: shape: (2, num_alignments)
+        the ID-based alignment in new IDs
+
+    :raises ValueError:
+        if the datatype of the alignment data frame is imcompatible (neither string nor integer).
+    """
+    # convert labels to IDs
+    for side, tf in zip(EA_SIDES, (left, right)):
+        if isinstance(tf, TriplesFactory) and is_string_dtype(alignment[side]):
+            logger.debug(f"Mapping label-based alignment for {side}")
+            # map labels, using -1 as fill-value for invalid labels
+            # we cannot drop them here, since the two columns need to stay aligned
+            alignment[side] = alignment[side].apply(tf.entity_to_id.get, -1)
+        if not is_numeric_dtype(alignment[side]):
+            raise ValueError(f"Invalid dype in alignment dataframe for side={side}: {alignment[side].dtype}")
+
+    # filter alignment
+    invalid_mask = (alignment.values < 0).any(axis=1) | (
+        alignment.values >= numpy.reshape(numpy.asarray([left.num_entities, right.num_entities]), newshape=(1, 2))
+    ).any(axis=1)
+    if invalid_mask.any():
+        logger.warning(
+            f"Dropping {format_relative_comparison(part=invalid_mask.sum(), total=alignment.shape[0])} "
+            f"alignments due to invalid labels.",
+        )
+        alignment = alignment.loc[~invalid_mask]
+
+    # map alignment from old IDs to new IDs
+    return torch.as_tensor(alignment.to_numpy().T, dtype=torch.long) + entity_offsets.view(1, 2)
```

Comment:
Shouldn't this be `entity_offsets.view(2,1)` ? Because otherwise I get a RuntimeError that the sizes don't match. Also `entity_offsets` should probably be cast to `long` right? Because this gave me trouble later on.

Reply:
`view` -> yes, fixed in fe9e4ec

`entity_offsets` should already be `long`, since it comes from the output of `cat_triples`.